### PR TITLE
Create Container component

### LINKED ISSUE #22076: [EXT] Create component: Container
We need to create a new Container component in Figma. This component should have a variant called `maxWidth` with three different sizes: Small (max: 360), Medium (max: 480), and Large (max: 720). Containers are used to constrain a content's width to the current breakpoint, while keeping it fluid. This component will allow us to maintain consistency in our design system and improve the efficiency of our design process.

[FigJam](https://www.figma.com/file/IXyyE5ZNzWrNhnIesQbmal/Page%2FGrid-Audit-and-Insight-Report?type=whiteboard&node-id=0%3A1&t=ApZPZAYpM8T72RNU-1)

### **Technical Details:**

1. **Component Name**: Container
2. **Description**: Containers are used to constrain a content's width to the current breakpoint, while keeping it fluid.
3. **Variant**: `maxWidth`
   - Small (max: 360)
   - Medium (max: 480)
   - Large (max: 720)

The component should be designed in a way that it can be easily resized and adapted to different contexts in our designs. It should also follow our existing design language and style guide.

### **Acceptance Criteria:**

1. The Container component is created in Figma.
2. The component has a `maxWidth` variant.
3. The `maxWidth` variant includes three sizes: Small (max: 360), Medium (max: 480), and Large (max: 720).
4. The component is flexible and can be easily adapted to different design contexts.
5. The component follows our existing design language and style guide.
6. The component is added to our Figma design system library and is available for use in designs.

### PR DESCRIPTION
**Pull request by garrettbear**
## **Description**
We need to create a new `Container` component in Figma. This component should have a variant called maxWidth with three different sizes: Small (max: 360), Medium (max: 480), and Large (max: 720). Containers are used to constrain a content's width to the current breakpoint, while keeping it fluid. This component will allow us to maintain consistency in our design system and improve the efficiency of our design process.

## **Related issues**

Fixes: #22076 

## **Manual testing steps**

1. Run locally
2. [Visit Container component storybook ](http://localhost:6006/?path=/docs/components-componentlibrary-container--docs)

### **Before**
n/a

### **After**
Screenshot 2023-12-06 at 4 15 37 PM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/component-library/container/README.mdx [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { Canvas, Controls } from '@storybook/addon-docs';
+import * as ContainerStories from './container.stories';
+
+# Container
+
+Containers are used to constrain a content's width to the current breakpoint, while keeping it fluid.
```
Comment (georgewrmarshall) [inline ui/components/component-library/container/README.mdx:6]:
```suggestion
Containers are used to constrain a content's width to the current breakpoint, while keeping it fluid
```

File: ui/components/component-library/container/README.mdx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+nunc, vitae aliquam nunc nunc eget nunc. Nullam aliquam, nisl eget
+aliquam ultrices, nunc nunc aliquam nunc, vitae aliquam nunc nunc eget
+nunc.
+</Container>
+<Container>
+No breakpoint: Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+Nullam aliquam, nisl eget aliquam ultrices, nunc nunc aliquam nunc,
+vitae aliquam nunc nunc eget nunc. Nullam aliquam, nisl eget aliquam
+ultrices, nunc nunc aliquam nunc, vitae aliquam nunc nunc eget nunc.
+</Container>
```
Comment (georgewrmarshall) [inline ui/components/component-library/container/README.mdx:57]: If you'd like to match code suggestion 

```suggestion
import { BackgroundColor } from '../../../helpers/constants/design-system';
import { Container, ContainerMaxWidth } from '../../component-library';

<Container
  maxWidth={ContainerMaxWidth.Sm}
  backgroundColor={BackgroundColor.primaryMuted}
  padding={4}
>
  Size "sm" container
</Container>
<Container
  maxWidth={ContainerMaxWidth.Md}
  backgroundColor={BackgroundColor.successMuted}
  padding={4}
>
  Size "md" container
</Container>
<Container
  maxWidth={ContainerMaxWidth.Lg}
  backgroundColor={BackgroundColor.warningMuted}
  padding={4}
>
  Size "lg" container
</Container>
<Container
  backgroundColor={BackgroundColor.errorMuted}
  padding={4}
>
  no max-width container
</Container>
```

File: ui/components/component-library/container/container.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+export default {
+  title: 'Components/ComponentLibrary/Container',
+  component: Container,
+  parameters: {
+    docs: {
+      page: README,
+    },
+  },
+  argTypes: {},
```
Comment (georgewrmarshall) [inline ui/components/component-library/container/container.stories.tsx:21]: Should we remove if not used?
```suggestion
```

File: ui/components/component-library/container/container.types.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  Lg = 'lg',
+}
+
+export interface ContainerStyleUtilityProps extends StyleUtilityProps {
+  /*
+   * Additional classNames to be added to the Container component
+   */
+  className?: string;
+  /**
+   * maxWidth prop sets the max-width of the Container component
```
Comment (georgewrmarshall) [inline ui/components/component-library/container/container.types.ts:18]: Do we want to add the max width sizes here for intellisense? 

```suggestion
   * maxWidth prop sets the max-width of the Container component
   * Sm (320px)
   * Md (480px)
   * Lg (720px)
```

**Review by georgewrmarshall — COMMENTED** [review]
Looking good!  A couple non-blocking suggestions. Would be good to use videos in the PR description to show manual testing steps for other reviewers

**Review by georgewrmarshall — DISMISSED** [review]
EXCELLENT

**Review by darkwing — DISMISSED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Review by DDDDDanica — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**